src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const operation of actions) {
+    switch (true) {
+      case operation.type === 'addProperties': {
+        Object.assign(state, operation.extraData);
+        break;
+      }
+
+      case operation.type === 'removeProperties': {
+        for (const property of operation.keysToRemove) {
+          delete state[property];
+        };
+        break;
+      }
+
+      case operation.type === 'clear': {
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+      }
+
+      default: {
+        break;
+      }
```

Comment:
you've not been told what to do in the default case, however, it's still better to throw an error